test/test-fs-event.c

```diff
@@ -153,7 +153,12 @@ static void fs_event_cb_del_dir(uv_fs_event_t* handle,
   ASSERT_PTR_EQ(handle, &fs_event);
   ASSERT_OK(status);
   ASSERT(events == UV_CHANGE || events == UV_RENAME);
+  /* There is a bug in the FreeBSD kernel where the filename is sometimes NULL.
+   * Refs: https://github.com/libuv/libuv/issues/4606
+   */
+  #ifndef __FreeBSD__
```

Comment:
Is it worth keeping the check on FreeBSD, but also allowing for `NULL`.

Reply:
Also, feel free to resolve the conversation either way. I always forget that unresolved comments prevent merging in this repo, even if the comments are non-blocking.

Reply:
No worries, that would make the code a bit cleaner.

Would you be ok w/ something like?

`ASSERT(filename == NULL || strcmp(filename, "watch_del_dir") == 0)`

And w/ that we avoid the `#ifndef` wording

Reply:
I don't think we should generalize the check if `NULL` is not correct in other cases.

Reply:
The kernel bug says it is unreliable, we may get the filename, or may not.

```cpp
#if defined(__FreeBSD__)
ASSERT(filename == NULL || strcmp(filename, "watch_del_dir") == 0)
#else
ASSERT_OK(strcmp(filename, "watch_del_dir"));
#endif
```

What about that?

Reply:
LGTM

Reply:
Thanks, Colin, I've addressed your comment.